### PR TITLE
Fix apply mage events due race condition

### LINKED ISSUE #32068: 'contentUpdated' event listener isn't always created due to a race condition
### Preconditions
1. Magento 2.3.6
1. Utilize a JavaScript bundling system (our system uses Advanced Bundling)
2. Utilize a theme based on theme-frontend-blank
3. Ensure your bundling system is loading the requirejs config in an element in the <head>
4. mage/mage should be loaded in the same file that applies the requirejs config

### Steps to reproduce
1. Load a webpage
2. Place a breakpoint on `$("body").on("contentUpdated", function() {`
2. Hard refresh the webpage

**NOTE**: This does not always work due to a race condition

### Expected result
1. `document.getElementsByTagName('body').length` is 1

### Actual result
1. In some cases, `document.getElementsByTagName('body').length` is not 1

### Technical Notes
In environments where there are fewer JavaScript files that can be loaded fast enough, `mage/mage` can be triggered quick to execute before a `<body>` tag exists in the DOM.

This is due to it being a dependency of, at least, `Magento_Theme/js/theme` in the blank theme - which is set as a global dependency by the `requirejs-config.js` file in that theme.

Previous ticket was reported in 2017: #10360 

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

When the code in lib/mage/mage.js is executed before the body tag is created. If the body tag doesn't exist, the $('body') return undefined and the event handler for the event 'contentUpdated' is not created.

So we should attach event to a parent element (html)
https://learn.jquery.com/events/event-delegation/

### Fixed Issues (if relevant)

1. Fixes magento/magento2#32068

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
Looks reasonable.
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved.

Failing tests look not related to changes from this PR.

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9126](https://jira.corp.magento.com/browse/ENGCOM-9126) has been created to process this Pull Request

**Review by Den4ik — APPROVED** [review]
✅ Good job.
@mrtuvn Thanks for your work

**Comment by magento-engcom-team** [comment]
Hi @Den4ik, thank you for the review.
[ENGCOM-9126](https://jira.corp.magento.com/browse/ENGCOM-9126) has been created to process this Pull Request

File: lib/web/mage/mage.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ define([
     /**
      * Init components inside of dynamically updated elements
      */
-    $('body').on('contentUpdated', function () {
```
Comment (tanghongtanghong) [inline lib/web/mage/mage.js:88]: test

File: lib/web/mage/mage.js [hunk 2]
```diff
@@ -85,7 +85,7 @@ define([
...
+    $('html').on('contentUpdated', 'body', function () {
```
Comment (tanghongtanghong) [inline lib/web/mage/mage.js]: 特特

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Functional Tests CE

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by Den4ik** [comment]
@magento run Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests EE

**Comment by engcom-Hotel** [comment]
:x: QA not Passed

Hello @mrtuvn,

We have checked this issue with the below steps:
Step 1: Use advance javascript bundling using below command:
php -f bin/magento config:set dev/js/enable_js_bundling 1
Step 2: Clean the cache
Step 3: Tried to open the Homepage (Theme : Luma)
Step 4: Apply the breakpoint in mage.js on L88 ($('html').on('contentUpdated', 'body', function () {)
Step 5: Hard refresh the page 17 times.

Note: Out of 10 times, almost 4 times the breakpoint was not hit. I am attaching a video for reference.


https://user-images.githubusercontent.com/51681618/128343670-b5671ca4-9520-4afa-8952-793e553bf8ea.mov

Thanks

**Comment by ihor-sviziev** [comment]
@mrtuvn, could you check the QA results?

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
Hi @engcom-Hotel,
Could you double-check if this issue reproducing with recent changes?

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-9126](https://jira.corp.magento.com/browse/ENGCOM-9126) has been created to process this Pull Request

**Comment by engcom-Dash** [comment]
❌ QA not Passed

**Manual Scenario Steps**

Step 1: Use advance javascript bundling using below command:
bin/magento config:set dev/js/enable_js_bundling 1
Step 2: Clean the cache
Step 3: Tried to open the Homepage (Theme : Luma)
Step 4: Apply the breakpoint in mage.js on L88 (document).on('contentUpdated', 'body', function () {
Step 5: Hard refresh the page many times.

**:heavy_check_mark: Expected result After Fix** 
-  Breakpoint will hit on every page reload

**:x: Actual result After Fix** 
-  Breaking point is not hitting every time

Attached is the screen recording for reference

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
As per [above-comment](https://github.com/magento/magento2/pull/33277#issuecomment-1011068148), still seeing the updated changes are not successfully solving the issue, moving it to _Changes Requested_.

**Comment by engcom-Lima** [comment]
Hi @mrtuvn, @ihor-sviziev
I have double checked the fix and it is working fine.
As per the [documentation](https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/js-bundling.html), JS bundling will only work in the **Production Mode**

Steps followed to verify the issue:-

1. Install fresh 2.4-develop instance of Magento
2. Enable JS bundling by following this [documentation](https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/themes/js-bundling.html) (Skip step 3 of minifying the JS files)
3.  Clean the cache
4. Open the Homepage (Theme : Luma)
5. Apply the breakpoint in mage.js (lib/mage/mage.js) on Line - 88 
```(document).on('contentUpdated', 'body', function () {```
6. Hard refresh the page many times.

**Result:**
Of all the 10 times, the breakpoint was hit every single time.

Based on this moving this PR to **Merge in Progress**

**Pull request merged**